presto-main/src/main/java/com/facebook/presto/metadata/FunctionListBuilder.java

```diff
@@ -60,7 +61,10 @@ public FunctionListBuilder window(String name, Class<? extends ValueWindowFuncti
                 parseTypeSignature(typeVariable),
                 Arrays.asList(argumentTypes).stream().map(TypeSignature::parseTypeSignature).collect(toImmutableList()),
                 false);
-        boolean canIgnoreNulls = name.equalsIgnoreCase("lag") || name.equalsIgnoreCase("lead");
+
+        WindowFunctionOptions[] options = clazz.getAnnotationsByType(WindowFunctionOptions.class);
+        boolean canIgnoreNulls = options.length > 0 && options[0].canIgnoreNulls();
```

Comment:
Please add `checkState(options.length == 0 || options.length == 1)` because you don't want to support multiple annotations of `WindowFunctionOptions` here. Even if you wanted to, you'd need different `canIgnoreNulls` logic.